src/modules/QtQml/Component.js

```diff
@@ -38,6 +38,10 @@ QMLComponent.prototype.createObject = function(parent, properties, componentCont
     engine.$basePath = bp;
 
     engine.operationState = oldState;
+
+    if (item.hasOwnProperty("parent") && parent) // Seems to be an Item. TODO: Use real inheritance and ask using instanceof.
```

Comment:
Note, the "TODO" comment here was copied from `QMLItem.dataChanged`. I'm not sure if there's now a better way to do this that I should use.